### PR TITLE
[PLACES-479] Add sort by date to record tables

### PR DESCRIPTION
**Pull request by iamdcj**
The user can now sort records by the various dates

**Trace**
Screenshot 2020-07-10 at 14 29 19 (2)

**Publish**
Screenshot 2020-07-10 at 14 29 24 (2)

[Demo link](https://drive.google.com/file/d/1zxQ6Nckir9lSAteIeTtyDbcvgfMhdTdj/view)

[JIRA ticket](https://pathcheck.atlassian.net/browse/PLACES-479)

### CODE REVIEW

File: src/components/_shared/RecordsTable/useSortBy.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { useState } from 'react';
+
+import moment from 'moment';
+
+export const useSortByDate = array => {
+  const [sortBy, setSortBy] = useState('NEWEST');
```
Comment (yonatanmk) [inline src/components/_shared/RecordsTable/useSortBy.js:6]: I would use an enum here

**Comment by iamdcj** [comment]
> Currently we seem to be making the assumption that sorting by newest Last Saved or Expires date does the same thing

They do the same thing. I am tempted to drop the sorting on the expiration date as the designs only show sorting on 'Last Saved'.

File: src/components/_shared/RecordsTable/useSortBy.js (hunk 1 above)
Comment (iamdcj) [reply src/components/_shared/RecordsTable/useSortBy.js:6]: @yonatanmk please can you provide a snippet/update the code - when you mentioned enums outside of TS I'm thinking;

`
const sortByOptions = {
 newest: 'NEWEST'
 ...
}
`

Comment (yonatanmk) [reply src/components/_shared/RecordsTable/useSortBy.js:6]: I'm just not a fan of string comparisons. I would just go with something like
```
const sortEnum = Object.freeze({
  NEWEST: 'NEWEST',
  OLDEST: 'OLDEST'
})
```

and use that throughout the components

**Pull request merged**